src/components/BottomMenuModal/HistoryModal/TransactionInfo.tsx

```diff
@@ -0,0 +1,203 @@
+      localStorage.getItem('latestUserOpStatus') ?? undefined;
+    const storedHash =
+      localStorage.getItem('latestTransactionHash') ?? undefined;
+    const storedInfo = localStorage.getItem('latestUserOpInfo') ?? undefined;
+    const storedChainId =
+      localStorage.getItem('latestUserOpChainId') ?? undefined;
+
+    setDisplayStatus(storedStatus);
+    setDisplayHash(storedHash);
+    setDisplayInfo(storedInfo);
+    setDisplayChainId(storedChainId);
+  }, []);
+
+  // Update the localStorage only when values change
+  useEffect(() => {
+    if (
+      userOpStatus &&
+      userOpStatus !== localStorage.getItem('latestUserOpStatus')
+    ) {
+      localStorage.setItem('latestUserOpStatus', userOpStatus);
+      setDisplayStatus(userOpStatus);
+    }
+
+    if (
+      transactionHash &&
+      transactionHash !== localStorage.getItem('latestTransactionHash')
+    ) {
+      localStorage.setItem('latestTransactionHash', transactionHash);
+      setDisplayHash(transactionHash);
+    }
+
+    if (
+      latestUserOpInfo &&
+      latestUserOpInfo !== localStorage.getItem('latestUserOpInfo')
+    ) {
+      localStorage.setItem('latestUserOpInfo', latestUserOpInfo);
+      setDisplayInfo(latestUserOpInfo);
+    }
+
+    if (
+      latestUserOpChainId &&
+      latestUserOpChainId.toString() !==
+        localStorage.getItem('latestUserOpChainId')
+    ) {
+      localStorage.setItem(
+        'latestUserOpChainId',
+        latestUserOpChainId.toString()
+      );
+      setDisplayChainId(latestUserOpChainId.toString());
+    }
+  }, [userOpStatus, transactionHash, latestUserOpInfo, latestUserOpChainId]);
+
+  const steps = ['Sending', 'Sent', 'Confirmed/Failed'] as const;
+
+  const stepIndex = (() => {
+    if (displayStatus === 'Sending') return 0;
+    if (displayStatus === 'Sent') return 1;
+    if (displayStatus === 'Confirmed' || displayStatus === 'Failed') return 2;
+    return -1;
+  })();
+
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => setCopied(false), 3000);
+      return () => clearTimeout(timer);
+    }
+
+    return undefined;
+  }, [copied]);
+
+  return (
+    <div className="flex flex-col items-center w-full max-w-md mx-auto">
+      {/* Progress Bar */}
+      <div className="flex w-full mb-2">
+        {steps.map((_, index) => {
+          const isStepCompleted = index <= stepIndex;
+          return (
+            <div
+              key={index}
+              className={`flex-1 h-2 mx-1 rounded ${
+                isStepCompleted
+                  ? displayStatus === 'Failed' && index === 2
+                    ? 'bg-[#FF366C]'
+                    : 'bg-medium_purple'
+                  : 'bg-white/[.5]'
+              }`}
+            />
+          );
+        })}
+      </div>
+      {/* UserOp status */}
+      <p className="text-sm font-medium text-light_purple">
+        Status:{' '}
+        <span className="font-normal text-white">{displayStatus ?? 'N/A'}</span>
```

Comment:
I think if we have no display status available - display something like "Starting..." or something similar, as "N/A" gives off a negative effect